pkgs/applications/graphics/unigine_heaven/default.nix

```
@@ -0,0 +1,66 @@
+{ lib,
+  stdenv,
+  fetchurl,
+  makeWrapper,
+  libX11,
+  libXext,
+  libXrandr,
+  freetype,
+  fontconfig,
+  libXrender,
+  libXinerama,
+  autoPatchelfHook,
+  libGL,
+  openal,
+}:
+
+let
+  version = "4.0";
+in
+
+stdenv.mkDerivation {
+  name = "Unigine_Heaven";
+  pname = "Unigine_Heaven";
+
+  src = fetchurl {
+    url = "https://assets.unigine.com/d/Unigine_Heaven-${version}.run";
+    sha256 = "19rndwwxnb9k2nw9h004hyrmr419471s0fp25yzvvc6rkd521c0v";
+  };
+
+  unpackPhase = "true";
+
+  opl = openal; #need for wrapProgram
+  lgl = libGL;
+
+  installPhase = ''
+    cp $src extract.run
+    chmod +x ./extract.run
+    ./extract.run --target $out
+    wrapProgram $out/heaven --prefix LD_LIBRARY_PATH : $libGL/lib:$out/bin:$opl/lib --run "cd $out"
```

Just refer to the packages directly, no need to create extra variables for them:

```suggestion
    wrapProgram $out/heaven --prefix LD_LIBRARY_PATH : ${libglvnd}/lib:$out/bin:${openal}/lib --run "cd $out"
```